jetbrains-core/src/software/aws/toolkits/jetbrains/ui/feedback/FeedbackDialog.kt

```diff
@@ -32,14 +37,20 @@ class FeedbackDialog(project: Project) : DialogWrapper(project) {
 
             val sentiment = panel.sentiment ?: throw IllegalStateException("sentiment was null after validation")
             val comment = panel.comment ?: throw IllegalStateException("comment was null after validation")
-            ApplicationManager.getApplication().executeOnPooledThread {
+            GlobalScope.launch(ApplicationThreadPool) {
+                val edtDispatcher = dispatcherFor(ModalityState.stateForComponent(panel.panel))
                 try {
                     TelemetryService.getInstance().sendFeedback(sentiment, comment)
```

Comment:
What do we want the behaviour here to be? Do we really want to block the UI on the actual post to the metrics service? Or should we adopt a more asynchronous pattern - they click "Send" we attempt to post the feedback, if it doesn't work they get a toast that says feedback failed, maybe offers them a chance to re-try?

Reply:
I think it makes more sense to take the small time penalty of blocking until success/fail than giving the impression of success and making the user context switch back into the feedback dialog. This matches our current behavior for create S3 bucket/IAM policy, though those tie back to a larger wizard-type form

Reply:
Seems reasonable